pex/third_party/__init__.py

```diff
@@ -116,10 +116,11 @@ def iter_root_packages(self, relpath):
       yield package
 
   def _filter_names(self, relpath, pattern, group):
-    pat = re.compile(r'^{prefix}{pattern}$'
-                     .format(prefix=self.prefix + ((relpath + os.sep) if relpath else ''),
-                             pattern=pattern))
-
+    # We use '/' here instead of os.sep because the zip file format spec specifies that paths must
+    # use forward slashes.
+    # See section 4.4.17 of https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
```

Comment:
Nit, can you please move the start of this comment to the prior line, as they are related? And add a period to the end of this sentence.

Thanks!